### PR TITLE
Blog app controllers spec.

### PR DESCRIPTION
**Pull request by firstpj**
In this milestone, I accomplished the following tasks:

Integrated RSpec into your project as a gem.
Generated a Request spec file encompassing all controllers.
Verified the correctness of each action by ensuring:
The response status was accurate.
The appropriate template was rendered.
The response body incorporated the correct placeholder text.
Updated the project's Readme documentation.
Resolved issues identified by the linter.

### CODE REVIEW

**Review by Davidosky007 — COMMENTED** [review]
Hi @firstpj 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has failing linter checks , kindly solve the issue

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -1 +1,185 @@
...
+<!-- Features -->
+
+### Key Features <a name="key-features"></a>
+
+> The key features of this project includes.
+
+- **Display Users**
+- **Display Posts**
+- **Display Likes on Posts**
+- **Display Comments on Posts**
```
Comment (Abenezer-Tilahun) [inline README.md]: - [ ] Kindly make sure that your **Key feature section of the README file describe the current project milestone .**

File: README.md [hunk 2]
```diff
@@ -1 +1,185 @@
...
+
+Example command:
+
+```sh
+  rails s
+```
+
+### Run tests
+
+- Tests are not available yet, please stay tuned for it.
```
Comment (Abenezer-Tilahun) [inline README.md]: - [ ] This Project milestone is related with test cases for your blog app So kindly mention  here how to run the test cases locally on machine.

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @firstpj 

Great job for implemented  the some of the Project Requirement  , but there are some changes that need to be implemented in order to proceed to the next project 👍

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README
✔️ Good Commit Messages
✔️ Correctly Setup and configure RSpecas a gem to your project..
✔️ Correctly Create a Request spec file for all your controllers.
✔️ Good job for Implemented Blog app - controllers specs.
✔️ Kindly make sure that All Test are Passing. Some of your App sample test cases are failing So kindly Fix the issues.

image

almost

## Required Changes ♻️
_Check the comments below the review._


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @firstpj, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Updated the README file ✅ 


## Required Changes ♻️

- [ ] You did great in terms of updating the README file, but your previous code reviewer asked to alter your `controller` test cases because they were failing last time and instead of updating or correcting the issue related to `controllers` testing, you deleted the whole folder `requests` from the `spec`. As per the milestone `requirement`, you should implement the tests for the controller `user` and `post` and make sure that your tests are passing before asking for a re-review for your project. Kindly fix the problem and try to bring back those files deleted by you and implement the things as expected.

Screenshot from 2024-01-26 15-50-05
Screenshot from 2024-01-26 15-48-18


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    it 'renders the index template' do
+      expect(response).to render_template(:index)
+    end
+
+    it 'includes correct placeholder text in the response body' do
+      expect(response.body).to include('Here is a list of posts for given user')
+    end
+  end
+
```
Comment (Olamarx) [inline spec/requests/posts_spec.rb]: - [ ] As you can see in the image below, two of your tests are failing, I will encourage to run all tests and ensure they are passing before making another submission
image

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @firstpj, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Linters passed ✔️
- Gitflow followed ✔️
- Meaningful commit messages ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /users/732/posts' do
+    it 'returns a 200 OK status' do
+      get '/users/732/posts'
+      expect(response).to have_http_status(200)
+    end
```
Comment (1GORDON) [inline spec/requests/posts_spec.rb:8]: Great job @firstpj 🤝
- As requested by the previous reviewer, kindly ensure all tests are passing and are all in green. At the moment one  test is still failing. Kindly fix this and resubmit your project for approval.
fail-test

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @firstpj ✋

Good job so far!

## Highlights!
- Implemented requested changes ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb (hunk 4 above)
Comment (firstpj) [reply spec/requests/posts_spec.rb:8]: Hello, thank you for the review, But I noticed that all the test are passing on my pc. here is the screenshot of my test output: Screenshot (408)

Comment (rloterh) [reply spec/requests/posts_spec.rb:8]: well resolved! ✔️

**Review by rloterh — COMMENTED** [review]
Hi Collins, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Kindly ensure to add a professional ReadMe to this project via [this link](https://github.com/microverseinc/readme-template), which should comply with the following:
- Ensure that provided placeholder texts/ guide instructions are removed and/or replaced with appropriate contents applicable under each section of your ReadMe. Similarly, kindly also remove guide notes for _license_ creation under the _license_ section.
- All main headers from the [ReadMe template](https://github.com/microverseinc/readme-template) should be present in your ReadMe, only the _FAQ_ section is optional.
- Ensure all headers have appropriate content below describing them.
- The _Future Features_ section of your ReadMe includes 1-3 potential features to be added later. 
- Remove the Live demo link from the ReadMe if the app is not yet deployed.
- Ensure to remove the Microverse template logo. Optionally, you can replace it with your own logo if you have one.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Olamarx** [comment]
Hello @firstpj and @rloterh, this is just a personal comment because I got the email notification of invalidating the project. There is a github glitch issue at the moment as I have experienced also in other reviews. What solved it is making the feature branch the default branch and you will be good to go. @rloterh you will notice all the readme content is actually there when you click on the readme file as you can see in the image below. 
image

**Review by Olamarx — APPROVED** [review]
Hi @firstpj, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Linters passed ✔️
- Readme file professional ✔️
- Gitflow followed ✔️

### Note, I am approving this project because the readme file issue is a github glitch and I have explained how it could be solved in a comment on this thread. Also, I reached you on Slack, you are not available, kindly always avail yourself when you make submissions.
### Congrats, your tests passed
image

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**